libs/aws/langchain_aws/vectorstores/inmemorydb/base.py

```diff
@@ -0,0 +1,1493 @@
+    MemoryDB creates the underlying vector schema. This is useful
+    for production use cases where you want to optimize the
+    vector schema for your use case. ex. using HNSW instead of
+    FLAT (knn) which is the default
+
+        .. code-block:: python
+
+            vector_schema = {
+                "algorithm": "HNSW"
+            }
+
+            rds = InMemoryVectorStore.from_texts(
+                texts, # a list of strings
+                metadata, # a list of metadata dicts
+                embeddings, # an Embeddings object
+                vector_schema=vector_schema,
+                redis_url="redis://localhost:6379",
+            )
+
+    Custom index schema can be supplied to change the way that the
+    metadata is indexed. This is useful for you would like to use the
+    hybrid querying (filtering) capability of MemoryDB.
+
+    By default, this implementation will automatically generate the index
+    schema according to the following rules:
+        - All strings are indexed as text fields
+        - All numbers are indexed as numeric fields
+        - All lists of strings are indexed as tag fields (joined by
+            langchain_aws.vectorstores.inmemorydb.constants.INMEMORYDB_TAG_SEPARATOR)
+        - All None values are not indexed but still stored in MemoryDB these are
+            not retrievable through the interface here, but the raw MemoryDB client
+            can be used to retrieve them.
+        - All other types are not indexed
+
+    To override these rules, you can pass in a custom index schema like the following
+
+        .. code-block:: yaml
+
+            tag:
+                - name: credit_score
+            text:
+                - name: user
+                - name: job
+
+    Typically, the ``credit_score`` field would be a text field since it's a string,
+    however, we can override this behavior by specifying the field type as shown with
+    the yaml config (can also be a dictionary) above and the code below.
+
+        .. code-block:: python
+
+            rds = InMemoryVectorStore.from_texts(
+                texts, # a list of strings
+                metadata, # a list of metadata dicts
+                embeddings, # an Embeddings object
+                index_schema="path/to/index_schema.yaml", # can also be a dictionary
+                redis_url="redis://localhost:6379",
+            )
+
+    When connecting to an existing index where a custom schema has been applied, it's
+    important to pass in the same schema to the ``from_existing_index`` method.
+    Otherwise, the schema for newly added samples will be incorrect and metadata
+    will not be returned.
+
+    """
+
+    DEFAULT_VECTOR_SCHEMA = {
+        "name": "content_vector",
+        "algorithm": "FLAT",
+        "dims": 1536,
+        "distance_metric": "COSINE",
+        "datatype": "FLOAT32",
+    }
+
+    def __init__(
+        self,
+        redis_url: str,
```

Comment:
Should we use `endpoint` or `host` here, since this will be the cluster endpoint? Consequently, we will need other optional inputs to construct the redis url, for example port, ssl etc with defaults.

Reply:
url will be the endpoint of the cluster.

Reply:
What I meant is that this will accept only "redis://host:port" at the moment, not the host, port, separately as parameters.